contracts/implementation/PoolKeeper.sol

```diff
@@ -137,18 +131,29 @@ contract PoolKeeper is IPoolKeeper, Ownable {
         try ILeveragedPool(_pool).poolUpkeep(lastExecutionPrice, latestPrice) {
             executionPrice[_pool] = latestPrice;
             // If poolUpkeep is successful, refund the keeper for their gas costs
-            uint256 gasSpent = startGas - gasleft();
-            try IOracleWrapper(ILeveragedPool(_pool).settlementEthOracle()).poll() {} catch Error(
-                string memory reason
-            ) {
-                emit PoolUpkeepError(_pool, reason);
-            }
-            payKeeper(_pool, gasPrice, gasSpent, savedPreviousUpdatedTimestamp, updateInterval);
             emit UpkeepSuccessful(_pool, data, lastExecutionPrice, latestPrice);
         } catch Error(string memory reason) {
             // If poolUpkeep fails for any other reason, emit event
             emit PoolUpkeepError(_pool, reason);
         }
+
+        uint256 gasSpent = startGas - gasleft();
+        uint256 reward;
+        // Emit events depending on whether or not the reward was actually paid
+        if (
+            IKeeperRewards(keeperRewards).payKeeper(
```

Comment:
should this get the keeper from the Pool contract instead?? Assuming you could drop the setKeeper from this contract as well?

Reply:
What do you mean by "get the keeper from the Pool contract"?
As in, get `msg.sender` from inside a `LeveragedPool` context?

Reply:
no, here its initialising the keeper with keeperRewards from PoolKeeper.sol, wouldnt you want `ILeveragedPool(_pool).keeperRewards`

Reply:
or you could drop the `setKeeperRewards` and `keeperRewards` LeveragedPool.sol, the double up is confusing me

Reply:
but maybe there is no way around it

Reply:
Oooh you just made me remember a dream last night where I removed `LeveragedPool::keeperRewards` and just called `PoolKeeper.keeperRewards()` in `onlyKeeperRewards`. Similar to what you're saying but kind of flipped the other way. This way we don't have to update every pool when we change `keeperRewards`.